hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -177,6 +182,25 @@ private TokenAssociation validateAndBuildAutoAssociation(
             validateTrue(validAssociations, NO_REMAINING_AUTOMATIC_ASSOCIATIONS);
             validateFalse(token.hasKycKey(), ACCOUNT_KYC_NOT_GRANTED_FOR_TOKEN);
             validateFalse(token.accountsFrozenByDefault(), ACCOUNT_FROZEN_FOR_TOKEN);
+
+            final var unlimitedAutoAssociations =
+                    config.getConfigData(EntitiesConfig.class).unlimitedAutoAssociationsEnabled();
+            if (unlimitedAutoAssociations) {
+                final var topLevelPayer = handleContext.payer();
+                final var syntheticCreation = TransactionBody.newBuilder()
+                        .tokenAssociate(
+                                new Builder().account(account.accountId()).tokens(token.tokenId()));
+                final var fees = handleContext.dispatchComputeFees(
+                        syntheticCreation.build(), topLevelPayer, ComputeDispatchFeesAsTopLevel.NO);
```

Comment:
I believe this has to be updated because we do not want to dispatch transactions in such cases.

Reply:
@netopyr, Could you please elaborate on that? We call only the `computeFees` method so that the calculation will be performed the same way as in a regular token associate transaction. This way we get the same 0.05$ per association and we add them to the crypto transfer fees. Also, there is no code duplication with this approach.

Reply:
We should not dispatch transactions within other transactions because this creates undesired dependencies. Instead, common functionality should be extracted into methods that can be called directly.